src/init/projectInit.ts

```diff
@@ -0,0 +1,152 @@
+import * as os from 'os';
+import * as path from 'path';
+import { get } from 'https';
+
+import * as fs from 'fs-extra';
+
+import { ExtensionContext, Disposable, commands, window, Uri, workspace, QuickPickItem, Progress } from 'vscode';
+
+import AdmZip = require('adm-zip');
+
+import { VSCodeCommands, GaugeCommands, GaugeVSCodeCommands, GAUGE_TEMPLATE_URL } from "../constants";
+import { FileListItem } from '../types/fileListItem';
+import { spawnSync, execSync, spawn } from 'child_process';
+
+export class ProjectInitializer extends Disposable {
+    private isGaugeInstalled: boolean;
+    private readonly _disposable: Disposable;
+
+    private readonly _templates: any = [
+        { name: 'python', desc: "template for gauge-python projects", },
+        { name: 'js', desc: "template for gauge-javascript projects", },
+        { name: 'ruby', desc: "template for gauge-ruby projects", },
+    ];
+
+    constructor(isGaugeInstalled: boolean) {
+        super(() => this.dispose());
+        this.isGaugeInstalled = isGaugeInstalled;
+        this._disposable = commands.registerCommand(GaugeVSCodeCommands.CreateProject, async () => {
+            await this.createProject();
+        });
+    }
+
+    dispose() {
+        this._disposable.dispose();
+    }
+
+    private async createProject() {
+        const tmpl = await window.showQuickPick(this.getTemplatesList());
+        if (!tmpl) return;
+        let options: any = { prompt: "Enter a name for your new project", placeHolder: "gauge-tests" };
+        const name = await window.showInputBox();
+        if (!name) return;
+        options = { canSelectFolders: true, openLabel: "Select a folder to create the project in" };
```

Comment:
use `canSelectMany` to enforce a single folder selection. Ref: [here](https://code.visualstudio.com/docs/extensionAPI/vscode-api#OpenDialogOptions)